native/cocos/renderer/gfx-agent/DeviceAgent.cpp

```diff
@@ -92,14 +93,20 @@ bool DeviceAgent::doInit(const DeviceInfo &info) {
 }
 
 void DeviceAgent::doDestroy() {
-    ENQUEUE_MESSAGE_1(
-        _mainMessageQueue, DeviceDestroy,
-        actor, _actor,
-        {
-            if (actor) {
-                actor->destroy();
-            }
-        });
+    if (!_mainMessageQueue) {
+        if(_actor) {
+            _actor->destroy();
```

Comment:
Does it need to set _actor to nullptr?

Reply:
No need. The actor is intrusive and can be deleted automatically.

Reply:
It is not IntrusivePtr now. It is a raw pointer. It is deleted in DeviceAgent destructor.